samcli/commands/init/init_templates.py

```diff
@@ -171,3 +173,46 @@ def is_dynamic_schemas_template(self, package_type, app_template, runtime, base_
             if option.get("appTemplate") == app_template:
                 return option.get("isDynamicTemplate", False)
         return False
+
+    def get_preprocessed_manifest(self):
+        """
+        This method get the manifest cloned from the git repo and preprocessed it.
+
+        Raises:
+            InvalidInitTemplateError: Exception raise when template is invalid or missing a parameter
+
+        Returns:
+            dict: This is preprocessed manifest with the use_case as key
+        """
+        manifest_dump = requests.get(self._manifest_url)
+        if not manifest_dump:
+            raise InvalidInitTemplateError(
+                "Can't retrieve templates manifest from {url}".format(url=self._manifest_url)
+            )
+        preprocessed_manifest = {}
+        manifest_body = manifest_dump.json()
+
+        for template_runtime in manifest_body:
+            template_list = manifest_body[template_runtime]
+            for template in template_list:
+                if "useCaseName" not in template:
```

Comment:
useCaseName is an additional field to existing dictionary within a list for a given runtime, so backwards compat should be okay here.

Could we add a comment on how the expected manifest structure needs to be like for this function?

Reply:
manifest structure? 
Do you mean the manifest from the aws-sam-cli-template repo or the preprocessed manifest that function creates?

Reply:
The manifest structure itself as referenced in https://github.com/aws/aws-sam-cli-app-templates/pull/90/files, the thing I have noticed is that we are using manifest structure that's in a different repo, but we dont have a comment explaining what exactly the structure should look like. Adding that will help with the readability of the code without having to jump to  a different repository.